dsgrid/config/project_config.py

```diff
@@ -166,6 +232,11 @@ class InputDatasetModel(DSGBaseModel):
         updateable=False,
         # TODO: add notes about warnings for outdated versions? DSGRID-189.
     )
+    required_dimensions: RequiredDimensionsModel = Field(
```

Comment:
Would it be better to define these at a level higher than dataset so that the requirements can be shared across datasets of the same category?

Reply:
The default option is here: define one set of requirements per dataset. The downside is that two datasets cannot share the same requirements.

Other options:
1. Define requirements using a field that exists in each dataset config. `data_source`, `sector_description`, some new term. The field must be an enum.
2. Allow the project coordinator to define this free-form. The list of requirements would be at the top level of the project config. Each set of requirements would have a name. Let's suppose one is called `residential`.  Each dataset in `datasets` would have an optional field called `dimension_requirements`. ResStock and NEMS Residential would have `residential.` If the field is not provided then all project dimension records must be present.

Reply:
We discussed and decided to leave under dataset for now, in part because some cases that look like reuse might actually involve some tweaks.